lib/jekyll/commands/serve/servlet.rb

```diff
@@ -21,10 +21,36 @@ def initialize(server, root, callbacks)
         # up with a different preference on which comes first.
 
         def search_file(req, res, basename)
-          # /file.* > /file/index.html > /file.html
+          # /file.* -> /file.html
```

Comment:
Any reason why? It was an intended change, since `search_file` _doesn't_ change "/file" into "/file/index.html" (that's `search_index_file`'s job)